### DISCUSSION TITLE
Make a branch follow working copy commit?

### DISCUSSION THREAD

**Discussion by johnwheeler** [jj-vcs/jj#2836] [Q&A] [answered]
Hi, what are the thoughts on an option where the branch follows the current commit (or even better get set to @-) after every commit?

I suppose I could write a shell script to do that, but I'm wondering if that is against the ethos of jj or something I am missing that it is bad to do that / prevents me from doing other things.

Thank you. I love jj

**Comment by martinvonz** [comment, marked as answer]
We have #2338 for letting you do `jj commit --advance-branches` (and presumably some short version of the flag, such as `-b`). We probably don't want a config option because then `jj new main; jj commit` would always move the `main` branch forward, which is probably (?) not what you always want.
